src/Twig/Extension/TemplateRegistryExtension.php

```diff
@@ -64,12 +65,12 @@ public function getPoolTemplate($name)
     }
 
     /**
+     * @param string $adminCode
+     *
      * @return AdminInterface|false|null
      */
-    private function getAdmin()
+    private function getAdmin($adminCode)
     {
-        return $this->pool->getAdminByAdminCode(
-            $this->requestStack->getCurrentRequest()->get('_sonata_admin')
```

Comment:
Should the requestStack be removed from the constructor then?

Reply:
Oh, yeah. D'oh. Good point.

Reply:
Either that or make the new argument optional

Reply:
I don't think that that's a good idea. The code will be more confusing and it's never used in the current templates.